admin-dev/themes/new-theme/js/pages/product/services/combinations-service.js

```diff
@@ -97,49 +66,46 @@ export default class CombinationsService {
   }
 
   /**
+   * @param {Number} productId
    * @param {Object} data Attributes indexed by attributeGroupId { 1: [23, 34], 3: [45, 52]}
    */
-  generateCombinations(data) {
+  generateCombinations(productId, data) {
     return $.ajax({
       url: this.router.generate('admin_products_combinations_generate', {
-        productId: this.productId,
+        productId,
       }),
       data,
       method: 'POST',
     });
   }
 
   /**
+   * It actually updates one combination at a time, but calls api designed for bulk update purpose
+   *  (same form applied to multiple combinations one by one)
+   *
+   * @param {number} combinationId
+   * @param {Object} data
+   *
    * @returns {Promise}
    */
-  getCombinationIds() {
-    return $.get(
-      this.router.generate('admin_products_combinations_ids', {
-        productId: this.productId,
-      }),
-    );
-  }
-
-  /**
-   * @param {string} orderBy
-   * @param {string} orderWay
-   */
-  setOrderBy(orderBy, orderWay) {
-    this.orderBy = orderBy;
-    this.orderWay = orderWay.toLowerCase() === 'desc' ? 'DESC' : 'ASC';
-  }
-
-  /**
-   * @returns {Object}
-   */
-  getFilters() {
-    return this.filters;
+  bulkUpdate(combinationId, data) {
+    return $.ajax({
+      url: this.router.generate('admin_products_combinations_bulk_edit_combination', {combinationId}),
+      data,
+      method: 'POST',
```

Comment:
Can't you force `PATCH` method here to solve your partial update problem with the controller?

Reply:
there is no data in request when PATCH is used here. Maybe jquery doesnt support it, i could try using plain js xhr